src/background/background.ts

```diff
@@ -11,38 +11,53 @@ chrome.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
     if (message.command === "startRecordingBytesTransferred") {
         IBytesRepository.instance.clearBytesTransferred();
 
-        chrome.debugger.attach({ tabId: tabId }, "1.2", () => {
-            chrome.debugger.sendCommand(
-                { tabId: tabId },
-                "Network.enable",
-                {},
-                () => {
-                    if (chrome.runtime.lastError) {
-                        console.error(chrome.runtime.lastError);
-                    }
-                }
-            );
+        chrome.tabs.get(tabId).then((tab) => {
+            if (tab.url && tab.url.startsWith("chrome://")) {
+                sendResponse({
+                    success: false,
+                    message:
+                        "Cannot calculate emissions for a chrome:// URL, e.g. manage extensions page. Please try again on a valid webpage.",
+                });
+            } else {
+                chrome.debugger.attach({ tabId: tabId }, "1.2").then(() => {
```

Comment:
random "1.2" ?

Reply:
It's the debugger protocol version - not sure what it does, just copied the example off the docs

![image](https://github.com/Theodo-UK/sustainability-calculator/assets/57725347/4646baf4-10cd-4296-8fc0-77bee00368b8)